envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/listeners/EnvoyListenersFactory.kt

```diff
@@ -313,10 +317,36 @@ class EnvoyListenersFactory(
                 .build()
     }
 
-    private fun accessLog(accessLogPath: String, accessLogType: String): AccessLog {
-        return AccessLog.newBuilder()
-                .setName("envoy.file_access_log")
-                .setTypedConfig(
+    private fun accessLog(
+        accessLogPath: String,
+        accessLogType: String,
+        accessLogFilterHttpCode: pl.allegro.tech.servicemesh.envoycontrol.groups.AccessLogFilter?
+    ): AccessLog {
+        val builder = AccessLog.newBuilder().setName("envoy.file_access_log")
+
+        accessLogFilterHttpCode?.let {
+            it.statusCodeFilter?.let {
+                builder.setFilter(
+                    AccessLogFilter.newBuilder().setStatusCodeFilter(
+                        io.envoyproxy.envoy.config.filter.accesslog.v2.StatusCodeFilter.newBuilder()
+                            .setComparison(
+                                io.envoyproxy.envoy.config.filter.accesslog.v2.ComparisonFilter.newBuilder()
+                                    .setOp(it.comparisonOP)
+                                    .setValue(
+                                        RuntimeUInt32.newBuilder()
+                                            .setDefaultValue(it.comparisonCode)
+                                            .setRuntimeKey("access_log_filter_http_code")
+                                            .build()
+                                    )
+                                    .build()
+                            )
+                            .build()
+                    )
+                )
+            }
+        }
```

Comment:
This is some christmas tree code :D How about extracting an innermost builder?
```suggestion
        accessLogFilterHttpCode?.let {
            it.statusCodeFilter?.let { statusCodeFilter ->
                builder.setFilter(accessLogFilterFrom(statusCodeFilter))
            }
        }
```

Reply:
done